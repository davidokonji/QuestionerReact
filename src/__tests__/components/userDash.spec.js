import React from 'react';
import { shallow } from 'enzyme';
import UserDash from '../../components/userDash';

let wrapper;
const props = {
  comment: 0,
  count: 0,
  upcoming: [[{}], [{}]],
  loading: true
};
describe('User Dashboard', () => {
  beforeAll(() => {
    wrapper = shallow(<UserDash {...props} />);
  });

  it('should render component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render when loading is false', () => {

    wrapper = shallow(<UserDash {...props} loading={false} />);

    expect(wrapper).toMatchSnapshot();
  });
});
