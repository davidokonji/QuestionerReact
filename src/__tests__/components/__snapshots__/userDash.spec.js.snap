// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Dashboard should render component correctly 1`] = `
<div
  className="col-12 col-md-8 mt-3"
>
  <div
    className="m-2"
  >
    <div
      className="top-bar row ml-2 ml-md-0"
    >
      <div
        className="col pl-md-5 pl-2 d-flex justify-content-center flex-column"
      >
        <h2
          className="align-self-center"
        >
          Question(s) posted
        </h2>
        <p
          className="align-self-center"
        >
          0
        </p>
      </div>
      <div
        className="col pl-md-5 pl-2 d-flex justify-content-center flex-column"
      >
        <h2
          className="align-self-center"
        >
          Question(s) commented
        </h2>
        <h3
          className="align-self-center"
        >
          0
        </h3>
      </div>
    </div>
    <div
      className="bottom-tabs mt-5 col-12 col-md-12"
    >
      <h3
        className="mt-5 mb-5"
      >
        Upcoming Meetups for you
      </h3>
      <div
        className="row ml-1 ml-md-0"
      >
        <div
          className="col-12 col-md-12"
        >
          <UpcomingLoader />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`User Dashboard should render when loading is false 1`] = `
<div
  className="col-12 col-md-8 mt-3"
>
  <div
    className="m-2"
  >
    <div
      className="top-bar row ml-2 ml-md-0"
    >
      <div
        className="col pl-md-5 pl-2 d-flex justify-content-center flex-column"
      >
        <h2
          className="align-self-center"
        >
          Question(s) posted
        </h2>
        <p
          className="align-self-center"
        >
          0
        </p>
      </div>
      <div
        className="col pl-md-5 pl-2 d-flex justify-content-center flex-column"
      >
        <h2
          className="align-self-center"
        >
          Question(s) commented
        </h2>
        <h3
          className="align-self-center"
        >
          0
        </h3>
      </div>
    </div>
    <div
      className="bottom-tabs mt-5 col-12 col-md-12"
    >
      <h3
        className="mt-5 mb-5"
      >
        Upcoming Meetups for you
      </h3>
      <div
        className="row ml-1 ml-md-0"
      >
        <PlainCard
          date="May 30 at 10:35 am"
        />
        <PlainCard
          date="May 30 at 10:35 am"
        />
      </div>
    </div>
  </div>
</div>
`;
